solutions/src/main/java/ru/sberbank/school/task10/ConcurrentScalablePool.java

```diff
@@ -0,0 +1,149 @@
+        this.threads = new ArrayBlockingQueue<>(max);
+        this.threadID = new ArrayBlockingQueue<>(max);
+        this.count = min;
+        this.lock = new ReentrantLock();
+        this.stopped = false;
+
+
+    }
+
+    @Override
+    public void start() {
+
+        lock.lock();
+        try {
+            if (!threads.isEmpty() && !stopped) {
+                throw new IllegalStateException("This method is already start!");
+            }
+            for (int i = 0; i < min; i++) {
+                threadID.add(i + 1);
+                newThread().start();
+            }
+        } finally {
+            lock.unlock();
+            stopped = false;
+        }
+
+    }
+
+    @Override
+    public void stopNow() {
+
+        if (stopped) {
+            throw new IllegalStateException("This method is already invoke!");
+        }
+
+        lock.lock();
+        try {
+
+            tasks.clear();
+
+            for (Thread t : threads) {
+                t.interrupt();
+            }
+            threads.clear();
+
+        } finally {
+            stopped = true;
+            lock.unlock();
+        }
+
+    }
+
+    @Override
+    public void execute(Runnable runnable) {
+
+        if (threads.isEmpty() && stopped) {
+            throw new NoSuchElementException("No threads running! Call start() first");
+        }
+
+
+        tasks.add(runnable);
+        lock.lock();
+        try {
+            if ((threads.size() < max) && (threadID.size() == 0)) {
+                if (count < max) {
+                    ++count;
+                    threadID.add(count);
+                }
+                newThread().start();
+            }
+        } finally {
+            lock.unlock();
+        }
+
+
+    }
+
+    private Thread newThread() {
+
+        int id = threadID.poll();
+
+        Thread thread = new Thread("Thread-" + id) {
+
+            @Override
+            public void run() {
+                Runnable runnable;
+                while (!Thread.interrupted()) {
+
+                    if (tasks.isEmpty()) {
+                        try {
+                            threadID.add(id);
+                            if (threads.size() > min) {
+                                threads.remove(Thread.currentThread());
+                                Thread.currentThread().interrupt();
+                            } else {
+                                tasks.wait();
+                            }
+
+                        } catch (InterruptedException e) {
+                            Thread.currentThread().interrupt();
+                        }
+                    }
+                    runnable = tasks.poll();
+
+                    try {
+                        runnable.run();
+                    } catch (NullPointerException ignored) {
+
+                    }
+                }
+            }
+        };
+        threads.add(thread);
+
+        return thread;
+    }
+
+    @Override
```

Comment:
Overload methods should not be split. Previous overloaded method located at line &#39;80&#39;.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a198546903>*